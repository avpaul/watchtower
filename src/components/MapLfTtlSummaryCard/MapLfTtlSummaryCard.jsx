import React from 'react';
import PropTypes from 'prop-types';
import Slider from 'react-slick';
import { carouselOptions } from '../../utils';
import LfTtlSummaryCard from '../LfTtlSummaryCard';

/**
 * @method MapLfTtlSummaryCard
 * @summary Method to render ttls/lfs cards together with their total number of fellows
 * @param { lfTtlSummary { Array of all lfs/ttls }
 * lfTtlCheckId {id of the lf/ttl card that is clicked } filterFellows {function}}
 * @returns A rendered version of a clickable lf/ttl card that displays the ttl/lf,
 * the number of fellows under the ttl/lf
 */
const MapLfTtlSummaryCard = ({ lfTtlSummary, filterFellows, lfTtlCheckId }) => (
  <Slider {...carouselOptions(4, 'contain slider')}>
    {lfTtlSummary.map(lfTtl => (
      <LfTtlSummaryCard
        id={`${lfTtl.id}`}
        key={lfTtl.id}
        picture={lfTtl.picture || ''}
        title={lfTtl.title || ''}
        name={lfTtl.name || ''}
        fellowsCount={lfTtl.fellowsCount}
        styles={lfTtl.styles}
        filterFellows={filterFellows}
        lfTtlCheckId={lfTtlCheckId}
      />
    ))}
  </Slider>
);

/**
 * @name MapLfTtlSummaryCard Proptypes
 * @type (Proptypes)
 * @property lfTtlSummary - Arrays of objects
 * @property filterFellows - function
 * @property lfTtlCheckId - mixed
 */
MapLfTtlSummaryCard.propTypes = {
  lfTtlSummary: PropTypes.arrayOf(PropTypes.shape()).isRequired,
  filterFellows: PropTypes.func.isRequired,
  lfTtlCheckId: PropTypes.oneOfType([PropTypes.string, PropTypes.number])
    .isRequired
};

export default MapLfTtlSummaryCard;
