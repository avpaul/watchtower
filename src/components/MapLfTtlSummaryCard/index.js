import MapLfTtlSummaryCard from './MapLfTtlSummaryCard';

export default MapLfTtlSummaryCard;
