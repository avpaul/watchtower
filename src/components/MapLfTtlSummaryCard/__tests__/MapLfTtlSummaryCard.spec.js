import React from 'react';
import { shallow } from 'enzyme';
import MapLfTtlSummaryCard from '../index';
import lfTtlSummaryMock from '../../../__mocks__/lfTtlSummary.json';

describe('Test fellowsSummaryCard snapshots', () => {
  const myMock = jest.fn();
  it('fellowSummaryCard should render properly', () => {
    expect(
      shallow(
        <MapLfTtlSummaryCard
          lfTtlSummary={lfTtlSummaryMock}
          filterFellows={myMock}
          lfTtlCheckId={21}
        />
      )
    ).toMatchSnapshot();
  });
});
