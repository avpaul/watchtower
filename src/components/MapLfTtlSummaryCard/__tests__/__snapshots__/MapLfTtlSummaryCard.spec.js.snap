// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test fellowsSummaryCard snapshots fellowSummaryCard should render properly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MapLfTtlSummaryCard
    filterFellows={[Function]}
    lfTtlCheckId={21}
    lfTtlSummary={
        Array [
            Object {
              "fellowsCount": 6,
              "id": "main",
              "styles": Object {
                "nameAvatarDisplayStyle": "d-none",
                "titleDisplayStyle": "",
              },
              "title": "All TTLs",
            },
            Object {
              "fellowsCount": 0,
              "id": 2,
              "name": "Kesiena Akpobome",
              "styles": Object {
                "nameAvatarDisplayStyle": "",
                "titleDisplayStyle": "d-none",
              },
            },
            Object {
              "fellowsCount": 0,
              "id": 6,
              "name": "Olaoluwa Faniyi",
              "styles": Object {
                "nameAvatarDisplayStyle": "",
                "titleDisplayStyle": "d-none",
              },
            },
            Object {
              "fellowsCount": 0,
              "id": 7,
              "name": "Olaolu Akinsete",
              "styles": Object {
                "nameAvatarDisplayStyle": "",
                "titleDisplayStyle": "d-none",
              },
            },
            Object {
              "fellowsCount": 0,
              "id": 10,
              "name": "Samuel Kubai",
              "styles": Object {
                "nameAvatarDisplayStyle": "",
                "titleDisplayStyle": "d-none",
              },
            },
            Object {
              "fellowsCount": 6,
              "id": 11,
              "name": "Trust  Birungi",
              "styles": Object {
                "nameAvatarDisplayStyle": "",
                "titleDisplayStyle": "d-none",
              },
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <LfTtlSummaryCard
          fellowsCount={6}
          filterFellows={[Function]}
          id="main"
          lfTtlCheckId={21}
          loading={false}
          name=""
          picture=""
          styles={
                    Object {
                              "nameAvatarDisplayStyle": "d-none",
                              "titleDisplayStyle": "",
                            }
          }
          title="All TTLs"
/>,
        <LfTtlSummaryCard
          fellowsCount={0}
          filterFellows={[Function]}
          id="2"
          lfTtlCheckId={21}
          loading={false}
          name="Kesiena Akpobome"
          picture=""
          styles={
                    Object {
                              "nameAvatarDisplayStyle": "",
                              "titleDisplayStyle": "d-none",
                            }
          }
          title=""
/>,
        <LfTtlSummaryCard
          fellowsCount={0}
          filterFellows={[Function]}
          id="6"
          lfTtlCheckId={21}
          loading={false}
          name="Olaoluwa Faniyi"
          picture=""
          styles={
                    Object {
                              "nameAvatarDisplayStyle": "",
                              "titleDisplayStyle": "d-none",
                            }
          }
          title=""
/>,
        <LfTtlSummaryCard
          fellowsCount={0}
          filterFellows={[Function]}
          id="7"
          lfTtlCheckId={21}
          loading={false}
          name="Olaolu Akinsete"
          picture=""
          styles={
                    Object {
                              "nameAvatarDisplayStyle": "",
                              "titleDisplayStyle": "d-none",
                            }
          }
          title=""
/>,
        <LfTtlSummaryCard
          fellowsCount={0}
          filterFellows={[Function]}
          id="10"
          lfTtlCheckId={21}
          loading={false}
          name="Samuel Kubai"
          picture=""
          styles={
                    Object {
                              "nameAvatarDisplayStyle": "",
                              "titleDisplayStyle": "d-none",
                            }
          }
          title=""
/>,
        <LfTtlSummaryCard
          fellowsCount={6}
          filterFellows={[Function]}
          id="11"
          lfTtlCheckId={21}
          loading={false}
          name="Trust  Birungi"
          picture=""
          styles={
                    Object {
                              "nameAvatarDisplayStyle": "",
                              "titleDisplayStyle": "d-none",
                            }
          }
          title=""
/>,
      ],
      "className": "contain",
      "infinite": false,
      "nextArrow": <NavArrow
        buttonClass="slick-next"
        handleChartClose="contain slider"
        iconClass="fa-angle-right"
        onClick={[Function]}
/>,
      "prevArrow": <NavArrow
        buttonClass="slick-prev"
        handleChartClose="contain slider"
        iconClass="fa-angle-left"
        onClick={[Function]}
/>,
      "responsive": Array [
        Object {
          "breakpoint": 1000,
          "settings": Object {
            "slidesToScroll": 1,
            "slidesToShow": 2,
          },
        },
        Object {
          "breakpoint": 700,
          "settings": Object {
            "slidesToScroll": 1,
            "slidesToShow": 1,
          },
        },
      ],
      "rows": 1,
      "slidesToScroll": 4,
      "slidesToShow": 4,
      "swipeToSlide": true,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "main",
        "nodeType": "function",
        "props": Object {
          "fellowsCount": 6,
          "filterFellows": [Function],
          "id": "main",
          "lfTtlCheckId": 21,
          "loading": false,
          "name": "",
          "picture": "",
          "styles": Object {
            "nameAvatarDisplayStyle": "d-none",
            "titleDisplayStyle": "",
          },
          "title": "All TTLs",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "2",
        "nodeType": "function",
        "props": Object {
          "fellowsCount": 0,
          "filterFellows": [Function],
          "id": "2",
          "lfTtlCheckId": 21,
          "loading": false,
          "name": "Kesiena Akpobome",
          "picture": "",
          "styles": Object {
            "nameAvatarDisplayStyle": "",
            "titleDisplayStyle": "d-none",
          },
          "title": "",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "6",
        "nodeType": "function",
        "props": Object {
          "fellowsCount": 0,
          "filterFellows": [Function],
          "id": "6",
          "lfTtlCheckId": 21,
          "loading": false,
          "name": "Olaoluwa Faniyi",
          "picture": "",
          "styles": Object {
            "nameAvatarDisplayStyle": "",
            "titleDisplayStyle": "d-none",
          },
          "title": "",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "7",
        "nodeType": "function",
        "props": Object {
          "fellowsCount": 0,
          "filterFellows": [Function],
          "id": "7",
          "lfTtlCheckId": 21,
          "loading": false,
          "name": "Olaolu Akinsete",
          "picture": "",
          "styles": Object {
            "nameAvatarDisplayStyle": "",
            "titleDisplayStyle": "d-none",
          },
          "title": "",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "10",
        "nodeType": "function",
        "props": Object {
          "fellowsCount": 0,
          "filterFellows": [Function],
          "id": "10",
          "lfTtlCheckId": 21,
          "loading": false,
          "name": "Samuel Kubai",
          "picture": "",
          "styles": Object {
            "nameAvatarDisplayStyle": "",
            "titleDisplayStyle": "d-none",
          },
          "title": "",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "11",
        "nodeType": "function",
        "props": Object {
          "fellowsCount": 6,
          "filterFellows": [Function],
          "id": "11",
          "lfTtlCheckId": 21,
          "loading": false,
          "name": "Trust  Birungi",
          "picture": "",
          "styles": Object {
            "nameAvatarDisplayStyle": "",
            "titleDisplayStyle": "d-none",
          },
          "title": "",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <LfTtlSummaryCard
            fellowsCount={6}
            filterFellows={[Function]}
            id="main"
            lfTtlCheckId={21}
            loading={false}
            name=""
            picture=""
            styles={
                        Object {
                                    "nameAvatarDisplayStyle": "d-none",
                                    "titleDisplayStyle": "",
                                  }
            }
            title="All TTLs"
/>,
          <LfTtlSummaryCard
            fellowsCount={0}
            filterFellows={[Function]}
            id="2"
            lfTtlCheckId={21}
            loading={false}
            name="Kesiena Akpobome"
            picture=""
            styles={
                        Object {
                                    "nameAvatarDisplayStyle": "",
                                    "titleDisplayStyle": "d-none",
                                  }
            }
            title=""
/>,
          <LfTtlSummaryCard
            fellowsCount={0}
            filterFellows={[Function]}
            id="6"
            lfTtlCheckId={21}
            loading={false}
            name="Olaoluwa Faniyi"
            picture=""
            styles={
                        Object {
                                    "nameAvatarDisplayStyle": "",
                                    "titleDisplayStyle": "d-none",
                                  }
            }
            title=""
/>,
          <LfTtlSummaryCard
            fellowsCount={0}
            filterFellows={[Function]}
            id="7"
            lfTtlCheckId={21}
            loading={false}
            name="Olaolu Akinsete"
            picture=""
            styles={
                        Object {
                                    "nameAvatarDisplayStyle": "",
                                    "titleDisplayStyle": "d-none",
                                  }
            }
            title=""
/>,
          <LfTtlSummaryCard
            fellowsCount={0}
            filterFellows={[Function]}
            id="10"
            lfTtlCheckId={21}
            loading={false}
            name="Samuel Kubai"
            picture=""
            styles={
                        Object {
                                    "nameAvatarDisplayStyle": "",
                                    "titleDisplayStyle": "d-none",
                                  }
            }
            title=""
/>,
          <LfTtlSummaryCard
            fellowsCount={6}
            filterFellows={[Function]}
            id="11"
            lfTtlCheckId={21}
            loading={false}
            name="Trust  Birungi"
            picture=""
            styles={
                        Object {
                                    "nameAvatarDisplayStyle": "",
                                    "titleDisplayStyle": "d-none",
                                  }
            }
            title=""
/>,
        ],
        "className": "contain",
        "infinite": false,
        "nextArrow": <NavArrow
          buttonClass="slick-next"
          handleChartClose="contain slider"
          iconClass="fa-angle-right"
          onClick={[Function]}
/>,
        "prevArrow": <NavArrow
          buttonClass="slick-prev"
          handleChartClose="contain slider"
          iconClass="fa-angle-left"
          onClick={[Function]}
/>,
        "responsive": Array [
          Object {
            "breakpoint": 1000,
            "settings": Object {
              "slidesToScroll": 1,
              "slidesToShow": 2,
            },
          },
          Object {
            "breakpoint": 700,
            "settings": Object {
              "slidesToScroll": 1,
              "slidesToShow": 1,
            },
          },
        ],
        "rows": 1,
        "slidesToScroll": 4,
        "slidesToShow": 4,
        "swipeToSlide": true,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "main",
          "nodeType": "function",
          "props": Object {
            "fellowsCount": 6,
            "filterFellows": [Function],
            "id": "main",
            "lfTtlCheckId": 21,
            "loading": false,
            "name": "",
            "picture": "",
            "styles": Object {
              "nameAvatarDisplayStyle": "d-none",
              "titleDisplayStyle": "",
            },
            "title": "All TTLs",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "2",
          "nodeType": "function",
          "props": Object {
            "fellowsCount": 0,
            "filterFellows": [Function],
            "id": "2",
            "lfTtlCheckId": 21,
            "loading": false,
            "name": "Kesiena Akpobome",
            "picture": "",
            "styles": Object {
              "nameAvatarDisplayStyle": "",
              "titleDisplayStyle": "d-none",
            },
            "title": "",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "6",
          "nodeType": "function",
          "props": Object {
            "fellowsCount": 0,
            "filterFellows": [Function],
            "id": "6",
            "lfTtlCheckId": 21,
            "loading": false,
            "name": "Olaoluwa Faniyi",
            "picture": "",
            "styles": Object {
              "nameAvatarDisplayStyle": "",
              "titleDisplayStyle": "d-none",
            },
            "title": "",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "7",
          "nodeType": "function",
          "props": Object {
            "fellowsCount": 0,
            "filterFellows": [Function],
            "id": "7",
            "lfTtlCheckId": 21,
            "loading": false,
            "name": "Olaolu Akinsete",
            "picture": "",
            "styles": Object {
              "nameAvatarDisplayStyle": "",
              "titleDisplayStyle": "d-none",
            },
            "title": "",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "10",
          "nodeType": "function",
          "props": Object {
            "fellowsCount": 0,
            "filterFellows": [Function],
            "id": "10",
            "lfTtlCheckId": 21,
            "loading": false,
            "name": "Samuel Kubai",
            "picture": "",
            "styles": Object {
              "nameAvatarDisplayStyle": "",
              "titleDisplayStyle": "d-none",
            },
            "title": "",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "11",
          "nodeType": "function",
          "props": Object {
            "fellowsCount": 6,
            "filterFellows": [Function],
            "id": "11",
            "lfTtlCheckId": 21,
            "loading": false,
            "name": "Trust  Birungi",
            "picture": "",
            "styles": Object {
              "nameAvatarDisplayStyle": "",
              "titleDisplayStyle": "d-none",
            },
            "title": "",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
